src/main/java/spoon/support/compiler/jdt/ReferenceBuilder.java

```diff
@@ -418,11 +421,20 @@ CtReference getDeclaringReferenceFromImports(char[] expectedName) {
 		ref.setType(this.<T>getTypeReference(messageSend.expectedType()));
 		if (messageSend.receiver.resolvedType == null) {
 			// It is crisis dude! static context, we don't have much more information.
-			if (messageSend.receiver instanceof SingleNameReference) {
-				ref.setDeclaringType(jdtTreeBuilder.getHelper().createTypeAccessNoClasspath((SingleNameReference) messageSend.receiver).getAccessedType());
-			} else if (messageSend.receiver instanceof QualifiedNameReference) {
-				ref.setDeclaringType(jdtTreeBuilder.getHelper().createTypeAccessNoClasspath((QualifiedNameReference) messageSend.receiver).getAccessedType());
-			}
+			try {
+				final JDTTreeBuilderHelper helper = jdtTreeBuilder.getHelper();
+				final CtTypeAccess<T> typeAccess;
+				if (messageSend.receiver instanceof SingleNameReference) {
+					typeAccess = helper.createTypeAccessNoClasspath((SingleNameReference) messageSend.receiver);
+				} else if (messageSend.receiver instanceof QualifiedNameReference) {
+					typeAccess = helper.createTypeAccessNoClasspath((QualifiedNameReference) messageSend.receiver);
+				} else if (messageSend.receiver instanceof ArrayReference) {
+					typeAccess = helper.createTypeAccessNoClasspath((ArrayReference) messageSend.receiver);
+				} else {
+					throw new SpoonException();
+				}
+				ref.setDeclaringType(typeAccess.getAccessedType());
+			} catch (final SpoonException e) { /* ignore */ }
```

Comment:
Since it is noclasspath mode any of these function may throw a `SpoonException` if they were not able to create a type access. Do you have a better idea?